ompi/mca/coll/han/coll_han_scatter.c

```diff
@@ -401,6 +401,9 @@ mca_coll_han_scatter_intra_simple(const void *sbuf, int scount,
         free(tmp_buf);
         tmp_buf = NULL;
     }
+    if ((NULL != reorder_buf) && (reorder_buf != sbuf)) {
```

Comment:
Isn't this enough?

```suggestion
    if (reorder_buf != sbuf) {
```

And we could remove the free of `reorder_buf` above.

Reply:
I checked man pages and this is safe, so I made the changes.